core/src/odbc_uri.rs

```diff
@@ -266,8 +272,21 @@ impl<'a> ODBCUri<'a> {
                 ServerAddress::parse(server).map_err(Error::InvalidClientOptions)?
             ])
             .credential(cred)
+            .app_name(self.get_app_name(None))
             .build())
     }
+
+    fn get_app_name(&mut self, uri: Option<&str>) -> Option<String> {
+        if let Some(uri) = uri {
+            APP_NAME_REGEX
+                .captures(uri)
+                .and_then(|cap| cap.name("appname").map(|s| s.as_str()))
+                .map_or(self.remove(&[APPNAME]), |s| Some(s))
+        } else {
+            self.remove(&[APPNAME])
+        }
+        .map(String::from)
```

Comment:
Should we default this to "odbc-driver" so that there is some telemetry for the raw odbc driver?

Reply:
Yeah, it used to be set to `mongo_odbc_driver` using [this variable](https://github.com/mongodb/mongo-odbc-driver/blob/7bcd73ce65d0dde61267d636739d420a96df2db9/core/src/conn.rs#L26).  Maybe we can use that name, or switch the underscores to dashes like with the connectors.

Reply:
This seems really out of place here since we have constants centrally located elsewhere. I'm asking Alexi what she wants to name to be and will consolidate that to `constants/src/lib.rs` and delete this variable.

Reply:
I got feedback from Alexi on the name and removed this variable since it wasn't being used anywhere.